services/frontend/pkg/revaconfig/config.go

```diff
@@ -9,7 +10,15 @@ import (
 )
 
 // FrontendConfigFromStruct will adapt an oCIS config struct into a reva mapstructure to start a reva service.
-func FrontendConfigFromStruct(cfg *config.Config) map[string]interface{} {
+func FrontendConfigFromStruct(cfg *config.Config) (map[string]interface{}, error) {
+
+	webURL, err := url.Parse(cfg.PublicURL)
+	if err != nil {
+		return map[string]interface{}{}, err
```

Comment:
Why not just `return nil, err`?
If it is because the caller might try to read from the map then `nil` is still fine since reads on a nil map are just like reads on an empty map.